pwndbg/commands/leakfind.py

```diff
@@ -0,0 +1,155 @@
+#Useful for debugging. Prints a map of child -> (parent, parent_start)
+def dbg_print_map(maps):
+    for child, parentInfo in maps.items():
+        print(hex(child) + "("+hex(parentInfo[0])+","+hex(parentInfo[1])+")")
+
+parser = argparse.ArgumentParser()
+parser.description = """
+Attempt to find a leak chain given a starting address. Scans memory near the given address, looks for pointers, 
+and continues that process to attempt to find leaks.\n
+Example: leakfind $rsp filename 0x48 6. This would look for any chains of leaks that point to a section in filename which begin near $rsp, are never 0x48 bytes further from a known pointer, 
+and are a maximum length of 6.\n
+"""
+parser.add_argument("address",help="Starting address to find a leak chain from.")
+parser.add_argument("-p","--page_name",type=str,nargs="?",default=None,help="Substring required to be part of the name of any found pages")
+parser.add_argument("-o","--max_offset",default=0x48,nargs="?",help="Max offset to add to addresses when looking for leak.")
+parser.add_argument("-d","--max_depth",default=0x4,nargs="?",help="Maximum depth to follow pointers to.")
+parser.add_argument("-s","--stride",nargs="?",default=0x1,help="Stride to add to add between pointers considered. For example, if this is 4 it would only consider pointers at an offset divisible by 4 from the starting pointer.")
+parser.add_argument('--negative_offset',nargs="?",default=0x0,help="Max negative offset to search before an address when looking for a leak.")
+@pwndbg.commands.ArgparsedCommand(parser)
+@pwndbg.commands.OnlyWhenRunning
+def leakfind(address=-1,page_name=None,max_offset=0x40,max_depth=0x4,stride=0x1,negative_offset=0x0):
+    if address == -1:
+        print("No starting address provided. Please run leakfind -h for more information.")
+        return
+
+    foundPages = getPage(address)
+
+    if not foundPages:
+        print("Starting address is not mapped. Please run leakfind -h for more information.")
+        return
+
+    if not pwndbg.memory.peek(address):
+        print("Unable to read from starting address. Please run leakfind -h for more information.")
+        return
+
+    max_depth=int(max_depth)
+    #Just warn the user that a large depth might be slow. Probably worth checking offset^depth < threshold. Do this when more benchmarking is established.
+    if max_depth > 8:
+        print("leakfind may take a while to run on larger depths.")
+    
+    stride = int(stride)
+    address=int(address)
+    max_offset = int(max_offset)
+    negative_offset=int(negative_offset)
+    
+    #The below map stores a map of child address->(parent_address,parent_start_address)
+    #In the above tuple, parent_address is the exact address with a pointer to the child adddress.
+    #parent_start_address is an address that a previous address pointed to.
+    #We need to store both so that we can nicely create our leak chain.
+    visitedMap = {}
+    visitedSet = set()
+    visitedSet.add(int(address))
```

Comment:
Can we apply some formatting changes here?
1) use underscore naming for local variables - e.g. `visited_set`, `address_queue`, `time_to_depth_increase`
2) put a space around operators e.g. `time_to_depth_increase = 0` (same in other cases)
3) this could be: `visited_set = {int(address)}` - you can create a set literal if you put some values around `{}` (otherwise it will be an empty dict)

Reply:
I think I fixed all of these now.